src/components/inputs/TextField.js

```diff
@@ -480,6 +497,7 @@ export default class TextField extends BaseInput {
     const {lineHeight, ...typographyStyle} = typography;
     const color = this.getStateColor(this.props.color || this.extractColorValue());
     const inputStyle = [
+      this.shouldDisplayRightButton() && {paddingRight: 33},
```

Comment:
The 33 is a random number? what if the image is wider?

Reply:
the image is limited to a certain size. The 33 is the image width + the gap between the image and the text

Reply:
Made it more explicit.